packages/accounts-base/accounts_server.js

```diff
@@ -525,11 +533,15 @@ Ap._initServerMethods = function () {
   };
 
   methods.logout = function () {
+    var user = accounts.users.findOne(this.userId)
```

Comment:
I see your concerns. But to my understanding, we have to track the `userId` somehow, since we will lost the value after `userId` set to null, right? We either have to duplicate the `if` statement at the beginning and avoid unnecessary database hits  or we have to assign the `userId` to a local variable and use it in the `if` statement. *(Assuming we should not call the hook before `userId` actually sets to null.)*